### PR TITLE
Rebase with lastest AutoValue

### PR DESCRIPTION
**Pull request by felipecsl**
This PR is pretty much the diff from [this PR](https://github.com/google/auto/pull/87) applied to the current HEAD of AutoValue, since that PR can't be auto-merged anymore (it's outdated)

### CODE REVIEW

**Comment by felipecsl** [comment]
Still pending a fix for the travis build and tests. I'll get to it later.

File: auto-parcel/src/main/java/auto/parcel/AutoParcel.java [hunk 1]
```diff
@@ -33,20 +34,55 @@
...
+
+    /**
+     * Specifies that the annotated method is a validation method. The method should be a non-private
+     * no-argument method in an AutoParcel class. It will be called by the {@code build()} method of
+     * the {@link Builder @AutoParcel.Builder} implementation, immediately after constructing the new
+     * object. It can throw an exception if the new object fails validation checks.
+     */
+    @Retention(RetentionPolicy.SOURCE)
+    @Target(ElementType.METHOD)
+    public @interface Validate {}
```
Comment (JakeWharton) [inline auto-parcel/src/main/java/auto/parcel/AutoParcel.java]: This file should be two-space indent.

File: auto-parcel/src/main/java/auto/parcel/AutoParcel.java [hunk 2]
```diff
@@ -19,9 +19,10 @@
 import java.lang.annotation.Target;
 
 /**
- * Specifies that <a href="http://goo.gl/Ter394">AutoParcel</a> should generate an implementation
- * class for the annotated abstract class, implementing the standard {@link Object} methods like
- * {@link Object#equals equals} to have conventional value semantics. A simple example: <pre>
+ * Specifies that <a href="https://github.com/google/auto/tree/master/value">AutoParcel</a> should
```
Comment (JakeWharton) [inline auto-parcel/src/main/java/auto/parcel/AutoParcel.java]: Link is pointing to the wrong place.

File: auto-parcel/src/main/java/auto/parcel/AutoParcel.java [hunk 3]
```diff
@@ -19,9 +19,10 @@
...
+ * generate an implementation class for the annotated abstract class, implementing the standard
+ * {@link Object} methods like {@link Object#equals equals} to have conventional value semantics. A
+ * simple example: <pre>
  *
  *   &#64;AutoParcel
  *   abstract class Person {
```
Comment (JakeWharton) [inline auto-parcel/src/main/java/auto/parcel/AutoParcel.java]: This class should implement `Parcelable` (pre-existing, but you're in here).

File: auto-parcel/src/main/java/auto/parcel/AutoParcel.java [hunk 4]
```diff
@@ -33,20 +34,55 @@
  *     abstract int id();
  *   }</pre>
  *
+ * @see <a href="https://github.com/google/auto/tree/master/value">AutoParcel User's Guide</a>
```
Comment (JakeWharton) [inline auto-parcel/src/main/java/auto/parcel/AutoParcel.java]: Also wrong link

File: auto-parcel/src/main/java/auto/parcel/AutoParcel.java [hunk 5]
```diff
@@ -33,20 +34,55 @@
...
\ No newline at end of file
+
+    /**
+     * Specifies that AutoParcel should generate an implementation of the annotated class or interface,
+     * to serve as a <i>builder</i> for the value-type class it is nested within. As a simple example,
+     * here is an alternative way to write the {@code Person} class mentioned in the {@link AutoParcel}
+     * example: <pre>
+     *
+     *   &#64;AutoParcel
+     *   abstract class Person {
```
Comment (JakeWharton) [inline auto-parcel/src/main/java/auto/parcel/AutoParcel.java]: should implement `Parcelable`

**Comment by JakeWharton** [comment]
Love it! :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

**Comment by frankiesardo** [comment]
Excellent job :fireworks: version `0.3` is deployed and should be accessible soon

**Comment by felipecsl** [comment]
Thank you @frankiesardo and @JakeWharton for your great contributions for OOS and Android ecosystem

**Pull request merged**